api/src/main/java/org/infinispan/api/common/events/KeyValueCreatedListener.java

```diff
@@ -0,0 +1,9 @@
+package org.infinispan.api.common.events;
+
+/**
+ *
+ * @since 13.0
+ **/
+public interface KeyValueCreatedListener<K, V> extends KeyValueListener<K, V> {
+   void onCreate(KeyValueEvent<K, V> event);
```

Comment:
All of these callback methods should return a `CompletionStage<Void>` so the user can do non blocking operations if necessary.